bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/ResourcesPlugin.java

```diff
@@ -416,12 +417,16 @@ public static ResourcesPlugin getPlugin() {
 	 * Returns the workspace. The workspace is not accessible after the resources
 	 * plug-in has shutdown.
 	 *
-	 * @return the workspace that was created by the single instance of this
-	 *   plug-in class.
+	 * <b>Hint:</b> Accessing the Workspace in a static way is prone to start-up
+	 * order problem, please consider using any of your favorite techniques
+	 * (Declarative Services, ServiceTracker, Blueprint, ...) instead.
```

Comment:
This needs some examples as code snippet.

Reply:
I have added some code sample to the top of the javadoc now.